sticker/src/input.rs

```diff
@@ -26,11 +26,47 @@ impl Embeddings {
 
 impl From<FiFuEmbeddings<VocabWrap, StorageWrap>> for Embeddings {
     fn from(embeddings: FiFuEmbeddings<VocabWrap, StorageWrap>) -> Self {
-        let mut unknown = Array1::zeros(embeddings.dims());
+        let mut unknown = match embeddings.storage() {
+            StorageWrap::QuantizedArray(_) | StorageWrap::MmapQuantizedArray(_) => {
+                let quantizer = match embeddings.storage() {
+                    StorageWrap::QuantizedArray(a) => a.quantizer(),
+                    StorageWrap::MmapQuantizedArray(a) => a.quantizer(),
+                    _ => unreachable!(),
+                };
+
```

Comment:
Couldn't this be written as a single `match` block returning `a.quantizer()`?

Reply:
Good question, I don't think so, the problem is that `StorageWrap::QuantizedArray(a) | StorageWrap::MmapQuantizedArray(a)` would have incompatible types for `a`. And of course, in the outer block we first need to distinguish quantized embeddings from other embeddings.

Reply:
I meant something like:

```Rust
let quantizer = match embeddings.storage() {
    StorageWrap::QuantizedArray(a) => a.quantizer()
    StorageWrap::MmapQuantizedArray(a) => a.quantizer()
    _ => // deal with other embedding types
};

// deal with quantized embeddings
```

Reply:
Ah, I understand, but then the l2 normalization code could not be shared. And it would interleave quantizer and non-quantizer code.